### PR TITLE
AsList interface

### PR DESCRIPTION
**Pull request by bschofield-va**
This introduces an AsList<T> interface that is implemented by all DSTU2, STU3, and R4 model objects.
This enhancement is intended to improve readability and reduce the amount of indentation required, particularly when building lists of one.

Tests and samples have been updated to use `AsList.asList()` where possible and `List.of(...)` when not.

Before:
```
        .reaction(
             asList(
                 AllergyIntolerance.Reaction.builder()
                     .manifestation(
                         asList(
                             CodeableConcept.builder()
                                 .coding(
                                     asList(
                                         Coding.builder()
                                             .display("Inflammation of Skin")
                                             .system("urn:oid:2.16.840.1.113883.6.233")
                                             .code("2000001")
                                             .build()))
                                 .text("Inflammation of Skin")
                                 .build()))
                     .certainty(AllergyIntolerance.Certainty.likely)
                     .build()))
 ```
 
 After:
 ```
         .reaction(
             AllergyIntolerance.Reaction.builder()
                 .manifestation(
                     CodeableConcept.builder()
                         .coding(
                             Coding.builder()
                                 .display("Inflammation of Skin")
                                 .system("urn:oid:2.16.840.1.113883.6.233")
                                 .code("2000001")
                                 .build()
                                 .asList())
                         .text("Inflammation of Skin")
                         .build()
                         .asList())
                 .certainty(AllergyIntolerance.Certainty.likely)
                 .build()
                 .asList())
 ```

### CODE REVIEW

**Comment by jblefkove-va** [comment]
This is my favorite kind of PR

File: argonaut-dstu2/src/main/java/gov/va/api/health/dstu2/api/resources/AllergyIntolerance.java [hunk 1]
```diff
@@ -47,7 +48,7 @@
     example =
         "${dstu2.allergyIntolerance:gov.va.api.health.dstu2.api.swaggerexamples"
             + ".SwaggerAllergyIntolerance#allergyIntolerance}")
-public class AllergyIntolerance implements Resource {
+public class AllergyIntolerance implements AsList<AllergyIntolerance>, Resource {
```
Comment (jhulbert-va) [inline argonaut-dstu2/src/main/java/gov/va/api/health/dstu2/api/resources/AllergyIntolerance.java:51]: `Bundle` got missed. Not sure it matters though.

File: us-core-r4/src/future/main/java/gov/va/api/health/r4/api/resources/ServiceRequest.java [hunk 2]
```diff
@@ -34,26 +34,21 @@
...
-  isGetterVisibility = JsonAutoDetect.Visibility.NONE
-)
+    fieldVisibility = JsonAutoDetect.Visibility.ANY,
+    isGetterVisibility = JsonAutoDetect.Visibility.NONE)
 @Schema(
-  description = "https://www.hl7.org/fhir/R4/servicerequest.html",
-  example = "${r4.serviceRequest:com.example.Example#example}"
-)
+    description = "https://www.hl7.org/fhir/R4/servicerequest.html",
+    example = "${r4.serviceRequest:com.example.Example#example}")
```
Comment (jhulbert-va) [inline us-core-r4/src/future/main/java/gov/va/api/health/r4/api/resources/ServiceRequest.java:41]: Oof

File: us-core-r4/src/future/main/java/gov/va/api/health/r4/api/resources/HealthcareService.java [hunk 3]
```diff
@@ -28,13 +28,11 @@
...
-  isGetterVisibility = JsonAutoDetect.Visibility.NONE
-)
+    fieldVisibility = JsonAutoDetect.Visibility.ANY,
+    isGetterVisibility = JsonAutoDetect.Visibility.NONE)
 @Schema(
-  description = "https://www.hl7.org/fhir/R4/healthcareservice.html",
-  example = "${r4.healthcareService:com.example.Example#example}"
-)
+    description = "https://www.hl7.org/fhir/R4/healthcareservice.html",
+    example = "${r4.healthcareService:com.example.Example#example}")
```
Comment (jhulbert-va) [inline us-core-r4/src/future/main/java/gov/va/api/health/r4/api/resources/HealthcareService.java:35]: Ick

File: us-core-r4/src/future/main/java/gov/va/api/health/r4/api/resources/Slot.java [hunk 4]
```diff
@@ -26,13 +26,11 @@
...
-  isGetterVisibility = JsonAutoDetect.Visibility.NONE
-)
+    fieldVisibility = JsonAutoDetect.Visibility.ANY,
+    isGetterVisibility = JsonAutoDetect.Visibility.NONE)
 @Schema(
-  description = "https://www.hl7.org/fhir/R4/slot.html",
-  example = "${r4.slot:com.example.Example#example}"
-)
+    description = "https://www.hl7.org/fhir/R4/slot.html",
+    example = "${r4.slot:com.example.Example#example}")
```
[the remark this answers is not in the document]
Comment (jblefkove-va) [reply us-core-r4/src/future/main/java/gov/va/api/health/r4/api/resources/Slot.java:33]: I ran the `/future/` files through the formatter and removed the remaining stray references to `Collections.singletonList()`... but I don't see value in fixing their example links right now

**Review by jhulbert-va — APPROVED** [review]

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**